File: app/src/main/java/com/example/techfix/data/dao/SparePartDao.java
package com.example.techfix.data.dao;
import androidx.room.*;
import com.example.techfix.data.entity.SparePart;
import java.util.List;
@Dao public interface SparePartDao {
 @Insert(onConflict=OnConflictStrategy.REPLACE) void insertPart(SparePart part);
 @Query("SELECT * FROM spare_parts WHERE branchId = :branchId") List<SparePart> getPartsByBranch(int branchId);
 @Query("SELECT * FROM spare_parts WHERE branchId = :branchId AND name = :name LIMIT 1") SparePart getPartByNameAndBranch(int branchId,String name);
 @Query("UPDATE spare_parts SET quantity = :newQuantity WHERE partId = :partId") void updateQuantity(int partId,int newQuantity);
}

File: app/src/main/java/com/example/techfix/data/dao/SparePartDao.kt
package com.example.techfix.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.techfix.data.entity.SparePart

@Dao
interface SparePartDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertPart(part: SparePart)

    @Query("SELECT * FROM spare_parts WHERE branchId = :branchId")
    suspend fun getPartsByBranch(branchId: Int): List<SparePart>

    @Query("SELECT * FROM spare_parts WHERE branchId = :branchId AND name = :name LIMIT 1")
    suspend fun getPartByNameAndBranch(branchId: Int, name: String): SparePart?

    @Query("UPDATE spare_parts SET quantity = :newQuantity WHERE partId = :partId")
    suspend fun updateQuantity(partId: Int, newQuantity: Int)
}